src/testsuite/arith128_test_i512.c

```diff
@@ -0,0 +1,1639 @@
+	CONST_VUINT128_QxW (0xE7FC7153, 0x329C57AE, 0xBFA3EDAC, 0x96E14F5D)
+    );
+
+const __VEC_U_512 vec512_ten512_1 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0xBC51FB2E, 0xB21A2F22, 0x1E25CFEA, 0x703ED321),
+	CONST_VUINT128_QxW (0xAA1DA1BF, 0x28F8733B, 0x4475B579, 0xC88976C1),
+	CONST_VUINT128_QxW (0x94E65747, 0x46C40513, 0xC31E1AD9, 0xB83A8A97),
+	CONST_VUINT128_QxW (0x5D96976F, 0x8F9546DC, 0x77F27267, 0xFC6CF801)
+    );
+
+const __VEC_U_512 vec512_ten512_0 = CONST_VINT512_Q
+    (
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0)
+    );
+
+const __VEC_U_512 vec512_ten1024_6 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0x00000000, 0x00000000, 0x00000000, 0x00000000),
+	CONST_VUINT128_QxW (0x00000000, 0x00000325, 0xD9D61A05, 0xD4305D94),
+	CONST_VUINT128_QxW (0x34F4A3C6, 0x2D433949, 0xAE6209D4, 0x926C3F5B),
+	CONST_VUINT128_QxW (0xD2DB49EF, 0x47187094, 0xC1A6970C, 0xA7E6BD2A)
+);
+
+const __VEC_U_512 vec512_ten1024_5 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0x73C55349, 0x36A8DE06, 0x1E8D4649, 0xF4F3235E),
+	CONST_VUINT128_QxW (0x005B8041, 0x1640114A, 0x88BC491B, 0x9FC4ED52),
+	CONST_VUINT128_QxW (0x0190FBA0, 0x35FAABA6, 0xC356E38A, 0x31B5653F),
+	CONST_VUINT128_QxW (0x44597583, 0x6CB0B6C9, 0x75A351A2, 0x8E4262CE)
+);
+
+const __VEC_U_512 vec512_ten1024_4 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0x3CE3A0B8, 0xDF68368A, 0xE26A7B7E, 0x976A3310),
+	CONST_VUINT128_QxW (0xFC8F1F90, 0x31EB0F66, 0x9A202882, 0x80BDA5A5),
+	CONST_VUINT128_QxW (0x80D98089, 0xDC1A47FE, 0x6B7595FB, 0x101A3616),
+	CONST_VUINT128_QxW (0xB6F4654B, 0x31FB6BFD, 0xF56DEEEC, 0xB1B896BC)
+);
+
+const __VEC_U_512 vec512_ten1024_3 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0x8FC51A16, 0xBF3FDEB3, 0xD814B505, 0xBA34C411),
+	CONST_VUINT128_QxW (0x8AD822A5, 0x1ABE1DE3, 0x045B7A74, 0x8E1042C4),
+	CONST_VUINT128_QxW (0x62BE695A, 0x9F9F2A07, 0xA7E89431, 0x922BBB9F),
+	CONST_VUINT128_QxW (0xC9635986, 0x1C5CD134, 0xF451218B, 0x65DC60D7)
+);
+
+const __VEC_U_512 vec512_ten1024_2 = CONST_VINT512_Q
+    (
+	CONST_VUINT128_QxW (0x233E55C7, 0x231D2B9C, 0x9FCE837D, 0x1E43F61F),
+	CONST_VUINT128_QxW (0x7DE16CFB, 0x896634EE, 0x0ED1440E, 0xCC2CD819),
+	CONST_VUINT128_QxW (0x4C7D1E1A, 0x140AC535, 0x15C51A88, 0x991C4E87),
+	CONST_VUINT128_QxW (0x1EC29F86, 0x6E7C215B, 0xF55B2B72, 0x2919F001)
+);
+
+const __VEC_U_512 vec512_ten1024_1 = CONST_VINT512_Q
+    (
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0)
+    );
+
+const __VEC_U_512 vec512_ten1024_0 = CONST_VINT512_Q
+    (
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0),
+      (vui128_t) ((unsigned __int128) 0)
+    );
+
+#define TIME_10_ITERATION 10
```

Comment:
This is the number of iterations for perf timing loops. This is a good number for running on real hardware. But for QEMU, Mambo, or valgrind, I set this to 1. Otherwise testing gets tedious.

I am not sure I need to explain all that

Reply:
Why would you put the constant _value_ in the _name_ of the constant??
Perhaps `#define TIMING_ITERATIONS 10` ?